reporter/mocha-allure/AllureReporter.js

```diff
@@ -147,8 +147,9 @@ module.exports = class AllureReporter {
         this.currentTest.stage = Stage.RUNNING;
 
         if (clearFilesForPreviousAttempt) {
+            let retry = test._currentRetry;
             const state = cy && cy.state().runnable;
-            if (state && state.prevAttempts && state.prevAttempts.length > 0) {
+            if (state && retry > 0) {
```

Comment:
in case `test._currentRetry` is used `state` is not needed at all.
Few points to mention:
 - common practice is to declare variables with const, just to avoid mutating it, as here we obviously will not need to mutate it.
 - no need to specify variable when it is used just once, so `if( test._currentRetry > 0 )` would be fine.